src/main/java/core/basesyntax/strategy/MyFileReaderImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MyFileReaderImpl implements MyFileReader {
+    public List<String> read(String filePath) {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading file: " + filePath, e);
```

Comment:
You've handled the IOException according to the checklist by rethrowing it as a RuntimeException with an informative message. Good job on following the exception handling guideline.